src/ci/docker/host-x86_64/dist-x86_64-linux/build-cmake.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+set -ex
+source shared.sh
+
+CMAKE=3.20.3
+curl -L https://github.com/Kitware/CMake/releases/download/v$CMAKE/cmake-$CMAKE.tar.gz | tar xzf -
+
+mkdir cmake-build
+cd cmake-build
+hide_output ../cmake-$CMAKE/configure --prefix=/rustroot
+hide_output make -j$(nproc)
+hide_output make install
+
+cd ..
+rm -rf cmake-build
+rm -rf cmake-$CMAKE
```

Comment:
Shouldn't this be in src/ci/scripts? We're going to need this at least on dist-i686-linux and on all images that have Ubuntu < 22.04 (unless we update them).

Reply:
The 32-bit runner uses the scripts from the x64 directory (https://github.com/rust-lang/rust/blob/eadf69a6c6edfe220fc5b1b659e46e271d75a3a1/src/ci/docker/host-x86_64/dist-i686-linux/Dockerfile#L53 :man_shrugging:). But good point that we should install CMake on more runners. Probably to all runners that were modified in https://github.com/rust-lang/rust/pull/95026?

Reply:
Should be all runners that use either `ubuntu:18.04` or `ubuntu:20.04`.

Reply:
Shouldn't use https://github.com/rust-lang/rust/blob/master/src/ci/docker/scripts/cmake.sh, only move version out?

Reply:
Yeah, we should bump the version in that script and use it.